buildenv/jenkins/jobs/builds/Build-Test-Any-Platform

```diff
@@ -0,0 +1,71 @@
+/*******************************************************************************
+ * Copyright (c) 2018, 2018 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+ARGS = ['SDK_VERSION', 'PLATFORM']
+
+node('master') {
+    timestamps {
+        try{
+            retry(5){
+                checkout scm
+            }
+
+            variableFile = load 'buildenv/jenkins/common/variables-functions'
+
+            if (!params.UPSTREAM_JOB_NAME && !params.UPSTREAM_JOB_NUMBER) {
+                variableFile.set_job_variables('build')
+                buildFile = load 'buildenv/jenkins/common/build'
+            }
+
+            if (run_tests()) {
+                variableFile.set_job_variables('test')
+                testFile = load 'buildenv/jenkins/common/test'
+            }
+        } finally {
+            try{
+                cleanWs()
+            } catch(e) {
+                variableFile.printStackTrace(e)
+            }
+        }
+    }
+}
+
+node("${NODE}") {
+    if (!run_tests()) {
+        // build only: compile, archive, clean up
+        buildFile.build_all()
+    } else {
+        if (params.UPSTREAM_JOB_NAME && params.UPSTREAM_JOB_NUMBER) {
+            // fetch resources from upstream job, run tests, archive results, clean up
+            testFile.test_all_with_fetch()
```

Comment:
This will not work because JAVA_BIN was set on master when the test file was loaded
https://github.com/eclipse/openj9/pull/2093/files#diff-b19f95b3c3121dd4c7d986c2633a9deaR41

Given #1450 is very close, I think we should not worry about covering both compile and test jobs with this change
We should however, probably still do the PRs in this file, since we haven't ironed out where that story is headed yet.

Reply:
Added a 2nd checkout for the test script